public_html/wp-content/plugins/pattern-directory/includes/notifications.php

```diff
@@ -0,0 +1,243 @@
+ */
+add_action( 'transition_post_status', __NAMESPACE__ . '\monitor_post_status_transitions', 20, 3 );
+
+/**
+ * Hook into post status transitions to detect changes that warrant a notification.
+ *
+ * @param string   $new_status
+ * @param string   $old_status
+ * @param \WP_Post $post
+ *
+ * @return void
+ */
+function monitor_post_status_transitions( $new_status, $old_status, $post ) {
+	if ( PATTERN !== get_post_type( $post ) ) {
+		return;
+	}
+
+	if ( 'publish' === $new_status && in_array( $old_status, array( 'pending', SPAM_STATUS ) ) ) {
+		notify_pattern_approved( $post );
+	} elseif (
+		SPAM_STATUS === $new_status
+		|| ( 'pending' === $new_status && 'publish' === $old_status )
+	) {
+		notify_pattern_flagged( $post );
+	} elseif ( UNLISTED_STATUS === $new_status ) {
+		notify_pattern_unlisted( $post );
+	}
+}
+
+/**
+ * Notify when a pattern has been approved.
+ *
+ * @param \WP_Post $post
+ *
+ * @return void
+ */
+function notify_pattern_approved( $post ) {
+	$author = get_user_by( 'id', $post->post_author );
+	if ( ! $author ) {
+		return;
+	}
+
+	$email  = $author->user_email;
+	$locale = get_user_locale( $author );
+
+	$pattern_title = get_the_title( $post );
+	$pattern_url = get_permalink( $post );
+
+	if ( $locale ) {
+		switch_to_locale( $locale );
+	}
+
+	$subject = esc_html__( 'Pattern published', 'wporg-patterns' );
+
+	$message = sprintf(
+		// translators: Plaintext email message. Note the line breaks. 1. Pattern title; 2. Pattern URL;
+		esc_html__( 'Hello!
+
+Thank you for submitting your pattern, %1$s. It is now live in the Block Pattern Directory!
+
+%2$s', 'wporg-patterns' ),
+		esc_html( html_entity_decode( $pattern_title ) ),
+		esc_url_raw( $pattern_url )
+	);
+
+	if ( $locale ) {
+		restore_current_locale();
+	}
+
+	send_email( $email, $subject, $message );
+}
+
+/**
+ * Notify when a pattern has been unpublished for review.
+ *
+ * @param \WP_Post $post
+ *
+ * @return void
+ */
+function notify_pattern_flagged( $post ) {
+	$author = get_user_by( 'id', $post->post_author );
+	if ( ! $author ) {
+		return;
+	}
+
+	$email  = $author->user_email;
+	$locale = get_user_locale( $author );
+
+	$pattern_title = get_the_title( $post );
+
+	if ( $locale ) {
+		switch_to_locale( $locale );
+	}
+
+	$flags = get_posts( array(
+		'post_type' => FLAG,
+		'post_parent' => $post->ID,
+		'post_status' => PENDING_STATUS,
+	) );
+	$reason = '';
+	if ( ! empty( $flags ) ) {
+		$reasons = array();
+		foreach ( $flags as $flag ) {
+			$terms = get_the_terms( $flag, REASON );
+			if ( is_array( $terms ) ) {
+				$reasons += $terms;
+			}
+		}
+		$reasons = array_map(
+			function( \WP_Term $reason ) {
+				return wp_strip_all_tags( $reason->description );
+			},
+			$reasons
+		);
+		$reasons = array_unique( $reasons );
+		$reason = trim( implode( "\n", $reasons ) );
+	} else {
+		// If it doesn't have flags, it must have gotten here by getting marked as spam.
+		$spam_term = get_term_by( 'slug', '4-spam' );
```

Comment:
I think you need the tax here. in my test, a spammed pattern sent the default reason, not the spam reason.

```suggestion
		$spam_term = get_term_by( 'slug', '4-spam', REASON );
```

Reply:
Bah, I had already committed this fix, but I forgot to push it.